modules/kubernetes-addons/main.tf

```diff
@@ -332,6 +332,8 @@ module "adot_collector_java" {
   amazon_prometheus_workspace_endpoint = var.amazon_prometheus_workspace_endpoint
   amazon_prometheus_workspace_region   = var.amazon_prometheus_workspace_region
   addon_context                        = local.addon_context
+
+  depends_on = [module.opentelemetry_operator]
```

Comment:
what happens when the operator is not installed/available first? will this eventually reconcile and therefore we can remove the `depends_on`? Same for the other collectors below